src/library/type_context.cpp

```diff
@@ -2415,8 +2415,7 @@ bool type_context_old::mk_nested_instance(expr const & m, expr const & m_type) {
 
         // HACK(gabriel): do not reuse type-class cache for nested resolution problems
         // For one example that easily breaks, see the default field values in `init/control/lawful.lean`
-        // TODO(gabriel): allow local caching
```

Comment:
What about the example that easily break, mentioned above this line?

Reply:
That's when you reuse the cache.  We don't do this here.  We start with an empty cache (as opposed to no caching at all).